### ISSUE TITLE
Need AllocateReadBuffer()

### ISSUE THREAD

**Issue by rodrigc** [opalmer/pywincffi#125]
In Twisted, I tried to replace pywin32 with pywincffi: https://github.com/twisted/twisted/pull/920

I found that winfile.AllocateReadBuffer() is not in pywincffi.

@opalmer can you add it?

**Comment by opalmer** [comment]
I don't think it's needed? That's not a standard win32 function and it appears to be a helper created for pywin32 as an abstraction. Twisted is calling `ReadFile()` and using `AllocateReadBuffer()` in `_win32serialport.py` to create a buffer to write the data to it appears. 

pywincffi also has the `ReadFile()` function but instead of requiring a buffer to write a result to, which does not mirror the win32 api signature, you need to tell it how much data to read and then `ReadFile()` should be able to return the read data.

**Comment by exvito** [comment]
Chiming in with a note:

From a quick read of Twisted's _win32serialport.py it seems to be using Overlapped I/O on ReadFile(), that's why it needs a buffer. I'm not sure the current ReadFile() implementation supports overlapped I/O, but #116 is still pending...

**Comment by opalmer** [comment]
Yeah I was looking at that.....`ReadFile()` may need some modification or we might need a new function like `ReadFileEx`. This honestly needs some more research first to figure out how to accomplish Overlapped I/O correctly with cffi. Most of the structures and things like `GetOverlappedResult` are there but building a test case/fix around win32serialport.py is probably what needs to happen just to make sure everything works as expected.

**Comment by opalmer** [comment]
So looking at the tests:

https://github.com/opalmer/pywincffi/blob/master/tests/test_kernel32/test_overlapped.py#L20

overlapped IO is possible right now just not through `ReadFile` like @exvito said. I'm going to look updating `ReadFile()` to support overrlapped IO like it's supposed to.

As for `AllocateReadBuffer`, I may or may not create a similar function for pywincffi. Reason being is looking at pywin32's source code AllocateReadBuffer appears to just return a standard python buffer object which is something pywincffi does not have an analog for (we're using cffi rather making a C-extension). Regardless, given we're working with _win32serialport here getting overlapped support into ReadFile is probably the thing that needs to be done to resolve this issue.
